### PR TITLE
[rush] added partial prerelease functionality

### PR DESCRIPTION
**Pull request by raygesualdo**
Solves #1135.

This PR adds a `--partial-prerelease` flag when running `rush publish` with the `--prerelease-name` flag. When the `--partial-prerelease` flag is used, prereleases will be publish _only for_ packages that have user-created changes (hotfix, patch, minor, major) as opposed to Rush changes (dependency). 

This allows for subsets of monorepo packages to be published as a single prerelease without A. publishing the all monorepo packages or B. requiring version policies to be utilized.

### CODE REVIEW

File: apps/rush-lib/src/logic/PublishUtilities.ts [hunk 1]
```diff
@@ -388,6 +388,10 @@ export class PublishUtilities {
           if (!depProject.shouldPublish || projectsToExclude && projectsToExclude.has(depName)) {
             // No version change.
             return;
+          } else if (prereleaseToken && prereleaseToken.hasValue && prereleaseToken.isPartialPrerelease && depChange.changeType! < ChangeType.hotfix) {
+            // FOr partial prereleases, do not version bump dependecies with the `prereleaseToken`
+            // value unless an actual change (hotfix, patch, minor, major) has occured
+            return
```
Comment (octogonz) [inline apps/rush-lib/src/logic/PublishUtilities.ts]: Tslint should be failing because of the missing semicolon here. If not, could you let us know?

Comment (raygesualdo) [reply apps/rush-lib/src/logic/PublishUtilities.ts]: It emitted a warning, but not an error. So the build still ran.

Comment (raygesualdo) [reply apps/rush-lib/src/logic/PublishUtilities.ts]: I fixed the warnings though.

**Comment by raygesualdo** [comment]
Added a changelog entry. Don't know why `rush change` added so many empty ones 😕

File: apps/rush-lib/src/logic/PublishUtilities.ts (hunk 1 above)
Comment (octogonz) [reply apps/rush-lib/src/logic/PublishUtilities.ts]: Cool

**Comment by octogonz** [comment]
> Added a changelog entry. Don't know why `rush change` added so many empty ones 😕

@raygesualdo Thanks for pointing this out.  I just debugged it.  It's a bug where `rush change` is not fetching changes for your **origin/master**.  I opened issue https://github.com/Microsoft/web-build-tools/issues/1177.

**Comment by octogonz** [comment]
This was published as Rush 5.6.3.

**Comment by raygesualdo** [comment]
Thanks @octogonz!

**Pull request merged**